lib/internal/util/inspect.js

```diff
@@ -1742,6 +1742,34 @@ function formatArrayBuffer(ctx, value) {
   return [`${ctx.stylize('[Uint8Contents]', 'special')}: <${str}>`];
 }
 
+function getParentName(ctx) {
+  try {
+    let parentName = '';
+    for (let parentIdx = ctx.seen.length - 2; parentIdx >= 0; parentIdx--) {
+      const value = ctx.seen[parentIdx + 1];
+
+      if (typeof ctx.seen[parentIdx] === 'object') {
+        const keys = ObjectKeys(ctx.seen[parentIdx]);
```

Comment:
This can be a very expensive operation. We do have all that information while iterating through the object, so it's possible to track it along other information.
It is actually also available during the comparison itself. It would require carful consideration when to use that information though, as it would also probably hide other meaningful information in case we would directly use that.

Reply:
This function is using already the information that got collected while traversing the object 😄 

`It is actually also available during the comparison itself.` what do you mean? which comparison?

Reply:
I implemented what you suggested, so when we are traversing the object in `inspect`, we are always populating a variable called `parentPath` while we push elements into `ctx.seen` . This is maybe clearer to be read but now it is executed every time we traverse the structure, not only when needed ( so when we are encountering an array ).

Why?

because doing it while traversing the object we don't know what value will be pushed next inside `ctx.seen`